### PR TITLE
chore: upgrade to Bazel 7 with Bazel 6 as secondary on CI

### PR DESCRIPTION
**Pull request by gregmagolan**
No longer testing against Bazel 5.

---

### Type of change

- Chore (any other change that doesn't affect source or test files, such as configuration)

### Test plan

- Covered by existing test cases

### CODE REVIEW

**Review by thesayyn — APPROVED** [review]

File: e2e/bundle/MODULE.bazel [hunk 1]
```diff
@@ -1,7 +1,7 @@
 "Bazel dependencies"
 
 bazel_dep(name = "aspect_rules_esbuild", version = "0.0.0", dev_dependency = True)
-bazel_dep(name = "aspect_bazel_lib", version = "1.39.1", dev_dependency = True)
+bazel_dep(name = "aspect_bazel_lib", version = "1.40.0", dev_dependency = True)
```
Comment (jbedard) [inline e2e/bundle/MODULE.bazel:4]: Why not `2.x`?

**Review by jbedard — APPROVED** [review]

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -11,61 +11,56 @@ on:
...
-      - id: macos
-        run: echo "os=macos-latest" >> $GITHUB_OUTPUT
-        # Only run on main branch (not PRs) to minimize macOS minutes (billed at 10X)
-        # https://docs.github.com/en/billing/managing-billing-for-github-actions/about-billing-for-github-actions#included-storage-and-minutes
-        if: ${{ github.ref == 'refs/heads/main' }}
-      - id: windows
-        run: echo "os=windows-latest" >> $GITHUB_OUTPUT
-        # Only run on main branch (or PR branches that contain 'windows') to minimize Windows minutes (billed at 2X)
+      - uses: actions/checkout@v4
+      - id: bazel-version
```
Comment (alexeagle) [inline .github/workflows/ci.yaml:25]: would you be okay splitting the GHA wizardry from the PR for upgrading Bazel and adding the presets?

Comment (gregmagolan) [reply .github/workflows/ci.yaml:25]: sure. I'll split it out. not sure which should go in first but I'll figure it out

File: e2e/bundle/MODULE.bazel (hunk 1 above)
Comment (gregmagolan) [reply e2e/bundle/MODULE.bazel]: Yup. Could be 2.2.0 in the e2e test for coverage of the latest. Updated.

File: .github/workflows/ci.yaml (hunk 2 above)
Comment (gregmagolan) [reply .github/workflows/ci.yaml:25]: Split out to https://github.com/aspect-build/rules_esbuild/pull/180

File: MODULE.bazel [hunk 3]
```diff
@@ -6,8 +6,8 @@ module(
     compatibility_level = 1,
 )
 
-bazel_dep(name = "aspect_bazel_lib", version = "1.39.1")
-bazel_dep(name = "aspect_rules_js", version = "1.34.1")
+bazel_dep(name = "aspect_bazel_lib", version = "1.40.0")
```
Comment (gregmagolan) [inline MODULE.bazel:9]: bazel-lib bump needed to pick up `is_bazel_7_or_greater()`

File: MODULE.bazel [hunk 4]
```diff
@@ -6,8 +6,8 @@ module(
...
+bazel_dep(name = "aspect_rules_js", version = "1.35.0")
```
Comment (gregmagolan) [inline MODULE.bazel:10]: 1.35.0 rules_js drops requirements for runfiles

**Pull request merged**